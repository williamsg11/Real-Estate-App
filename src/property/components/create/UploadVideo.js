/**
 * @flow
 */
import React, {Component, PropTypes} from 'react';
import {StyleSheet, TouchableHighlight, View} from 'react-native';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import colors from '../../../common/colors';
import ImagePicker from 'react-native-image-crop-picker';
import VideoPlayer from './../../../components/VideoPlayer';

export default class UploadVideo extends Component {
  static propTypes = {
    onFieldChange: PropTypes.func.isRequired,
    video: PropTypes.oneOfType([PropTypes.string, PropTypes.null]),
  };

  pickVideo = () => {
    const {onFieldChange} = this.props;
    ImagePicker.openPicker({
      smartAlbums: ['Videos'],
    }).then(media => {
      onFieldChange('video', media.path);
    });
  };

  removeVideo = video => {
    const {onFieldChange} = this.props;
    onFieldChange('video', null);
  };

  render() {
    const {header, footer, video} = this.props;

    return (
      <View style={styles.container}>

        {header}

        <TouchableHighlight
          style={styles.cameraIcon}
          onPress={() => this.pickVideo()}
          underlayColor="transparent">
          <FontAwesome name="video-camera" size={100} color={colors.white} />
        </TouchableHighlight>

        <View style={styles.menuContainer}>
          {video &&
            <VideoPlayer video={video} removeMedia={this.removeVideo} />}
        </View>

        {footer}

      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.smokeGreyLight,
  },
  contentContainer: {
    justifyContent: 'center',
    flexDirection: 'row',
    flexWrap: 'wrap',
  },
  menuContainer: {
    flex: 3,
    padding: 10,
    backgroundColor: 'white',
  },
  row: {
    justifyContent: 'center',
    padding: 5,
    margin: 10,
    width: 150,
    height: 150,
    backgroundColor: '#F6F6F6',
    alignItems: 'center',
    borderWidth: 0.5,
    borderRadius: 5,
    borderColor: '#CCC',
  },
  cameraIcon: {
    flex: 1,
    alignItems: 'center',
    paddingBottom: 20,
  },
  image: {
    width: 125,
    height: 125,
  },
  video: {
    flex: 1,
    height: 200,
    width: 300,
  },
});
