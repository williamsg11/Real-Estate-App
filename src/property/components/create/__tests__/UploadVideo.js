import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import UploadVideo from './../UploadVideo';

test('renders UploadVideo', () => {
  const tree = renderer
    .create(<UploadVideo onFieldChange={() => {}} video={null} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
