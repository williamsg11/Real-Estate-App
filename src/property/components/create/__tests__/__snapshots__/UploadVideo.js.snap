// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UploadVideo 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#e3e7ed",
      "flex": 1,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "alignItems": "center",
          "flex": 1,
          "paddingBottom": 20,
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 100,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      ï€½
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flex": 3,
        "padding": 10,
      }
    }
  />
</View>
`;
